bgbb_airflow/pred_airflow_job.py

```diff
@@ -10,19 +10,35 @@
 from bgbb_airflow.bgbb_utils import PythonLiteralOption
 
 
-def extract(spark, ho_start, model_win=90, sample_ids: List[int] = []):
+def pull_most_recent_params(spark):
+    pars_loc = (
+        "s3://net-mozaws-prod-us-west-2-pipeline-analysis/wbeard/bgbb_params/"
+    )
+    print("Reading params from {}".format(pars_loc))
+    spars = spark.read.parquet(pars_loc)
+    pars_df = spars.orderBy(spars.submission_date_s3.desc()).limit(1).toPandas()
+    print("Using params: \n{}".format(pars_df))
+    return pars_df
+
+
+def extract(spark, ho_start, model_win=90, sample_ids: Tuple[int] = ()):
     "TODO: increase ho_win to evaluate model performance"
     df, q = run_rec_freq_spk(
         ho_win=1,
         model_win=model_win,
         ho_start=ho_start,
-        sample_ids=sample_ids,
+        sample_ids=list(sample_ids),
         spark=spark,
     )
-    return df
+
+    # Hopefully not too far off from something like
+    # [0.825, 0.68, 0.0876, 1.385]
+    pars_df = pull_most_recent_params(spark=spark)
```

Comment:
Does this cause test failures? If it does, you can add something like the following:

```python
@pytest.fixture(autouse=True)
def mock_external_params(monkeypatch):
    def mocked_func(spark):
        # mocked data as a pandas dataframe using the value 
        raise NotImplementedError
    monkeypatch.setattr("bgbb_airflow.pred_airflow_job.pull_most_recent_params", mocked_func)
```
https://docs.pytest.org/en/3.0.1/monkeypatch.html

This will replace every call to the `pull_most_recent_params` in the tests with a dummy function.

Reply:
Good catch, I'd forgotten to run the tests but this worked